src/rest/rest_devices.c

```diff
@@ -250,28 +346,26 @@ int rest_devices_post_cb(const ulfius_req_t *req, ulfius_resp_t *resp, void *con
 
         if (database_prepare_array(jdatabase_list, rest->devicesList))
         {
-            ulfius_set_empty_body_response(resp, 500);
+            ulfius_set_json_body_response(resp, 500, j_post_resp);
```

Comment:
Why would you send a message body when server failure occurs?

Reply:
Because the inner list is already updated, and the device entry created. Not sending a response would result in a lost entry. Server failure at this point is not critical and it can recover from it. Database storage should be separated from REST and not something that the client should think about. From now on an error during writing to database file will only result in an error message being printed ant the REST client will not be informed.